### ISSUE TITLE
Properties throwing undefined on init event

### ISSUE THREAD

**Issue by decimoseptimo** [kenwheeler/slick#2549]
Perhaps init is firing too early

 ```js
$('.slider')
    .on('init', function(slick){
        alert(slick.initialSlide);
    })
    .slick();`

**Comment by leggomuhgreggo** [comment]
Hey can you make a reduced test case? There's a jsfiddle you can fork from the [contribution guidlines](https://github.com/kenwheeler/slick/blob/master/CONTRIBUTING.markdown).

Closing pending test case

**Comment by decimoseptimo** [comment]
I'm using custom controls, and I need access to `initialSlide` on init event but seems not to be an exposed property. I'm also testing 'currentSlide' which in this case seems not available on init event. http://jsfiddle.net/uoo9o22d/2/

**Comment by leggomuhgreggo** [comment]
check this out https://jsfiddle.net/5kv1mcL3/

**Comment by decimoseptimo** [comment]
Parameters were wrong, my bad.
I worked with flex slider before, and I wanted to give slick a try this
time.

I like to work with my own css classes and found the pagination controls
were harder to customize, there was a prototype function that wrapped
customPagination var in a LI element and whatnot, I found it easier to just
write my own pagination logic.

I'm not sure whether is actually harder or is it just me. But I feel like
flex exposes more stuff.
